spring-cloud-stream/src/main/java/org/springframework/cloud/stream/binder/SimpleExtendedPropertiesRegistry.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright 2016 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.stream.binder;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import org.springframework.beans.factory.BeanCreationException;
+
+/**
+ * @author Marius Bogoevici
+ */
+public class SimpleExtendedPropertiesRegistry<C, P> implements ExtendedPropertiesRegistry<C, P> {
+
+	private final Class<C> consumerPropertiesClass;
+
+	private final Class<P> producerPropertiesClass;
+
+	public SimpleExtendedPropertiesRegistry(Class<C> consumerPropertiesClass, Class<P> producerPropertiesClass) {
+		this.consumerPropertiesClass = consumerPropertiesClass;
+		this.producerPropertiesClass = producerPropertiesClass;
+	}
+
+	private Map<String, C> consumer = new HashMap<>();
+
+	private Map<String, P> producer = new HashMap<>();
+
+	public Map<String, C> getConsumer() {
+		return consumer;
+	}
+
+	public void setConsumer(Map<String, C> consumer) {
+		this.consumer = consumer;
+	}
+
+	public Map<String, P> getProducer() {
+		return producer;
+	}
+
+	public void setProducer(Map<String, P> producer) {
+		this.producer = producer;
+	}
+
+	public C getExtendedConsumerProperties(String channelName) {
```

Comment:
In this case it is really the channel, i.e. `input` vs `foo`. This would allow changing the attributes of the logical channel without needing to change anything about the destination.